src/syscheckd/seechanges.c

```diff
@@ -347,23 +361,41 @@ char *seechanges_addfile(const char *filename)
         goto cleanup;
     }
 
-    /* Run diff */
-    snprintf(
-        diff_cmd,
-        2048,
-        "diff \"%s\" \"%s\" > \"%s\" 2> /dev/null",
-        new_tmp,
-        old_tmp,
-        diff_tmp
-    );
+    if (is_nodiff((filename))) {
+        /* Dont leak sensible data with a diff hanging around */
+        snprintf(
+            diff_cmd,
+            2048,
+            "printf '<Diff truncated>' > \"%s\"",
```

Comment:
Also is printf always present on all supported platforms.  I really hate shelling out for this use case it's pointless outside of integrating into the reset of the function.